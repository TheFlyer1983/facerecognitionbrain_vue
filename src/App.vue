<script setup lang="ts">
// This starter template is using Vue 3 <script setup> SFCs
// Check out https://v3.vuejs.org/api/sfc-script-setup.html#sfc-script-setup
import particlesObject from '@/assets/particles/particles.json';
</script>

<template>
  <Particles id="tsparticles" class="particles" :options="particlesObject" />
  <img alt="Vue logo" src="./assets/logo.png" />
  <router-view />
</template>

<style>
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
  
}

.particles{
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: -1;
  background: linear-gradient(89deg, #FF5EDF 0%, #04C8DE 100%);
}
</style>
