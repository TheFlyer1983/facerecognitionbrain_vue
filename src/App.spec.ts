import { shallowMount, VueWrapper } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';

import App from './App.vue';

const pinia = createTestingPinia();

describe('Given the App', () => {
  const render = () =>
    shallowMount(App, {
      global: {
        plugins: [pinia],
        stubs: {
          'router-view': true,
          Particles: true
        }
      }
    });
  let wrapper: VueWrapper<InstanceType<typeof App>>;

  describe('when it is rendered', () => {
    beforeEach(() => {
      wrapper = render();
    });

    it('should match the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
