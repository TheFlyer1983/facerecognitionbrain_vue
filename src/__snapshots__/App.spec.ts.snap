// Vitest Snapshot v1

exports[`Given the App > when it is rendered > should match the snapshot 1`] = `
"<particles-stub id=\\"tsparticles\\" class=\\"particles\\" options=\\"[object Object]\\"></particles-stub>
<nav-stub></nav-stub>
<router-view-stub></router-view-stub>"
`;
